### ISSUE TITLE
[Windows] file_put_contents failed: folder does not exist inside a queue

### ISSUE THREAD

**Issue by Mordo95** [protonemedia/laravel-ffmpeg#278]
When the code is running using HLS rotating encryption on windows inside of a daemon queue, the first conversion works fine, but subsequent conversions all fail with the error message `ErrorException: file_put_contents(C:\Users\Mordo\AppData\Local\Temp/fcadda63f51de69f/3f3e68a945107db2.key): failed to open stream: No such file or directory in E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\EncryptsHLSSegments.php:160`.

I noticed that the same directory is used for all conversions (C:\Users\Mordo\AppData\Local\Temp/fcadda63f51de69f), but this one is deleted after the first conversion is done.

Operating system is Windows x64.

My job's handler function:
```
public function handle()
    {
        $lowBitrate = (new X264('aac'))->setKiloBitrate(250);
        $midBitrate = (new X264('aac'))->setKiloBitrate(500);
        $highBitrate = (new X264('aac'))->setKiloBitrate(1000);

        FFMpeg::fromDisk($this->disk)
            ->open($this->file)->exportForHLS()
            ->withRotatingEncryptionKey(function ($filename, $contents) {
                VideoKey::create([
                    'video_id' => $this->video->id,
                    'filename' => $filename,
                    'contents' => $contents
                ]);
            })
            ->addFormat($lowBitrate)
            ->addFormat($midBitrate)
            ->addFormat($highBitrate)
            ->toDisk('videos')
            ->save($this->video->id . '/index.m3u8');

        $media = FFMpeg::fromDisk($this->disk)
            ->open($this->file);

        FFMpeg::fromDisk($this->disk)
            ->open($this->file)
            ->getFrameFromSeconds($media->getDurationInSeconds() / 2)
            ->export()
            ->toDisk('videos')
            ->save($this->video->id . '/thumb.jpg');

        $this->video->status = "done";
        $this->video->save();
    }
```

full stack trace:
```
ErrorException: file_put_contents(C:\Users\Mordo\AppData\Local\Temp/fcadda63f51de69f/3f3e68a945107db2.key): failed to open stream: No such file or directory in E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\EncryptsHLSSegments.php:160
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'file_put_conten...', 'E:\\xampp\\htdocs...', 160, Array)
#1 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\EncryptsHLSSegments.php(160): file_put_contents('C:\\Users\\Mordo\\...', '\f\x86<\x90l\xC4\xE7\x17\x87\xE2B9\x18\x01\f...')
#2 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\EncryptsHLSSegments.php(225): ProtoneMedia\LaravelFFMpeg\Exporters\HLSExporter->rotateEncryptionKey()
#3 E:\xampp\htdocs\mywanderlust\vendor\evenement\evenement\src\Evenement\EventEmitterTrait.php(123): ProtoneMedia\LaravelFFMpeg\Exporters\HLSExporter->ProtoneMedia\LaravelFFMpeg\Exporters\{closure}('[hls @ 000001d7...', 'err')
#4 E:\xampp\htdocs\mywanderlust\vendor\alchemy\binary-driver\src\Alchemy\BinaryDriver\Listeners\Listeners.php(85): Evenement\EventEmitter->emit('listen', Array)
#5 E:\xampp\htdocs\mywanderlust\vendor\evenement\evenement\src\Evenement\EventEmitterTrait.php(123): Alchemy\BinaryDriver\Listeners\Listeners->Alchemy\BinaryDriver\Listeners\{closure}('[hls @ 000001d7...', 'err')
#6 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\FFMpeg\StdListener.php(49): Evenement\EventEmitter->emit('listen', Array)
#7 E:\xampp\htdocs\mywanderlust\vendor\alchemy\binary-driver\src\Alchemy\BinaryDriver\ProcessRunner.php(87): ProtoneMedia\LaravelFFMpeg\FFMpeg\StdListener->handle('err', 'ffmpeg version ...')
#8 E:\xampp\htdocs\mywanderlust\vendor\symfony\process\Process.php(1329): Alchemy\BinaryDriver\ProcessRunner->Alchemy\BinaryDriver\{closure}('err', 'ffmpeg version ...')
#9 E:\xampp\htdocs\mywanderlust\vendor\symfony\process\Process.php(1429): Symfony\Component\Process\Process->Symfony\Component\Process\{closure}('err', 'ffmpeg version ...')
#10 E:\xampp\htdocs\mywanderlust\vendor\symfony\process\Process.php(417): Symfony\Component\Process\Process->readPipes(true, false)
#11 E:\xampp\htdocs\mywanderlust\vendor\symfony\process\Process.php(239): Symfony\Component\Process\Process->wait()
#12 E:\xampp\htdocs\mywanderlust\vendor\alchemy\binary-driver\src\Alchemy\BinaryDriver\ProcessRunner.php(64): Symfony\Component\Process\Process->run(Object(Closure))
#13 E:\xampp\htdocs\mywanderlust\vendor\alchemy\binary-driver\src\Alchemy\BinaryDriver\AbstractBinary.php(207): Alchemy\BinaryDriver\ProcessRunner->run(Object(Symfony\Component\Process\Process), Object(SplObjectStorage), false)
#14 E:\xampp\htdocs\mywanderlust\vendor\alchemy\binary-driver\src\Alchemy\BinaryDriver\AbstractBinary.php(136): Alchemy\BinaryDriver\AbstractBinary->run(Object(Symfony\Component\Process\Process), false, Array)
#15 E:\xampp\htdocs\mywanderlust\vendor\php-ffmpeg\php-ffmpeg\src\FFMpeg\Media\AdvancedMedia.php(237): Alchemy\BinaryDriver\AbstractBinary->command(Array, false, Array)
#16 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Support\Traits\ForwardsCalls.php(23): FFMpeg\Media\AdvancedMedia->save()
#17 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Drivers\PHPFFMpeg.php(214): ProtoneMedia\LaravelFFMpeg\Drivers\PHPFFMpeg->forwardCallTo(Object(ProtoneMedia\LaravelFFMpeg\FFMpeg\AdvancedMedia), 'save', Array)
#18 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\MediaExporter.php(187): ProtoneMedia\LaravelFFMpeg\Drivers\PHPFFMpeg->__call('save', Array)
#19 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\MediaExporter.php(135): ProtoneMedia\LaravelFFMpeg\Exporters\MediaExporter->saveWithMappings()
#20 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\HLSExporter.php(209): ProtoneMedia\LaravelFFMpeg\Exporters\MediaExporter->save()
#21 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Collections\Traits\EnumeratesValues.php(692): ProtoneMedia\LaravelFFMpeg\Exporters\HLSExporter->ProtoneMedia\LaravelFFMpeg\Exporters\{closure}(Object(Illuminate\Support\Collection))
#22 E:\xampp\htdocs\mywanderlust\vendor\pbmedia\laravel-ffmpeg\src\Exporters\HLSExporter.php(223): Illuminate\Support\Collection->pipe(Object(Closure))
#23 E:\xampp\htdocs\mywanderlust\app\Jobs\VideoConverterJob.php(60): ProtoneMedia\LaravelFFMpeg\Exporters\HLSExporter->save('8/index.m3u8')
#24 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): App\Jobs\VideoConverterJob->handle()
#25 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#26 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#27 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#28 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\Container.php(610): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#29 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Bus\Dispatcher.php(128): Illuminate\Container\Container->call(Array)
#30 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(128): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(App\Jobs\VideoConverterJob))
#31 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Jobs\VideoConverterJob))
#32 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Bus\Dispatcher.php(132): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#33 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(118): Illuminate\Bus\Dispatcher->dispatchNow(Object(App\Jobs\VideoConverterJob), false)
#34 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(128): Illuminate\Queue\CallQueuedHandler->Illuminate\Queue\{closure}(Object(App\Jobs\VideoConverterJob))
#35 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Jobs\VideoConverterJob))
#36 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(120): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#37 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\CallQueuedHandler.php(70): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\Queue\Jobs\DatabaseJob), Object(App\Jobs\VideoConverterJob))
#38 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Jobs\Job.php(98): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\DatabaseJob), Array)
#39 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Worker.php(406): Illuminate\Queue\Jobs\Job->fire()
#40 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Worker.php(356): Illuminate\Queue\Worker->process('database', Object(Illuminate\Queue\Jobs\DatabaseJob), Object(Illuminate\Queue\WorkerOptions))
#41 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Worker.php(158): Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\DatabaseJob), 'database', Object(Illuminate\Queue\WorkerOptions))
#42 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Console\WorkCommand.php(116): Illuminate\Queue\Worker->daemon('database', 'default', Object(Illuminate\Queue\WorkerOptions))
#43 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Queue\Console\WorkCommand.php(100): Illuminate\Queue\Console\WorkCommand->runWorker('database', 'default')
#44 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): Illuminate\Queue\Console\WorkCommand->handle()
#45 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#46 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#47 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#48 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Container\Container.php(610): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#49 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Console\Command.php(136): Illuminate\Container\Container->call(Array)
#50 E:\xampp\htdocs\mywanderlust\vendor\symfony\console\Command\Command.php(255): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#51 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Console\Command.php(121): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#52 E:\xampp\htdocs\mywanderlust\vendor\symfony\console\Application.php(971): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#53 E:\xampp\htdocs\mywanderlust\vendor\symfony\console\Application.php(290): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#54 E:\xampp\htdocs\mywanderlust\vendor\symfony\console\Application.php(166): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#55 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Console\Application.php(93): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#56 E:\xampp\htdocs\mywanderlust\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php(129): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#57 E:\xampp\htdocs\mywanderlust\artisan(37): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#58 {main}```

**Comment by pascalbaljet** [comment]
Which version of the package are you running?
`composer show -- pbmedia/laravel-ffmpeg`

**Comment by Mordo95** [comment]
7.5.5

**Comment by danielcmm** [comment]
Same happens to me!

**Comment by serdarkok** [comment]
Has anyone been able to fix this problem?

**Comment by satriaajiputra** [comment]
Same here.  Happens for some videos. Even after error happened, the encoding process not stopped. Only exported the ``.ts`` files without the ``.key``

**Comment by pascalbaljet** [comment]
I just tagged [v7.5.8](https://github.com/protonemedia/laravel-ffmpeg/releases/tag/7.5.8) which should solve this problem. Could anyone confirm this fix works?

**Comment by serdarkok** [comment]
I'm gonna try now, just min.

**Comment by serdarkok** [comment]
I cleaned all cache, config, route after 7.5.8 updated.
Sorry, it doesn't work. There is no problem at during make transcode first video but next videos wants to first temporary folder which first created. Therefore, queue is throwing this error.

ErrorException: file_put_contents(/tmp/ec8dea8f1069a498/eb5659953308daf8.key): failed to open stream: No such file or directory in /media/serdarkok/LD/http/fflaravel/vendor/pbmedia/laravel-ffmpeg/src/Exporters/EncryptsHLSSegments.php:160....

**Comment by pascalbaljet** [comment]
Ah I see, I think I misunderstood the problem 😅

**Comment by serdarkok** [comment]
Hi pascal, when will be able to you fix this problem? I am looking forward this.

**Comment by pascalbaljet** [comment]
Fixed in v7.5.9 💪
